prime-router/src/main/kotlin/JurisdictionalFilters.kt

```diff
@@ -0,0 +1,95 @@
+import tech.tablesaw.selection.Selection
+
+/**
+ * A *JurisdictionalFilter* can be used in the jurisdictionalFilter property in an OrganizationService.
+ * It allowed you to create arbitrarily complex filters on data.
+ * Here is an example use:
+ *  `jurisdictionalFilter: { FilterByPatientOrFacilityLoc(AZ, Pima) }`
+ *
+ * The name `filterByPatientOrFacility` then maps via pseudo-reflection to an implementation of JurisdictionalFilter
+ * here.
+ *
+ * If you add a implementation here, you have to add it to the list of jurisdictionalFilters in Metadata.kt.
+ *
+ * A JurisdictionFilter is stateless.   It has a name property, which should be used in the filter definition -
+ * basically a simple way of implementing Reflection.
+ *
+ * Currently JurisdictionFilter implements its filtering by just re-using the very rich `Selection` functionality already in tablesaw.
+ *
+ * Hoping we implement some geospatial searches someday.
+ *
+ */
+interface JurisdictionalFilter {
+    /**
+     * Name of the filter function
+     */
+    val name: String
+
+    /**
+     * @args values passed to the filter
+     *
+     * @return the Selection object to be used to filter the
+     * tablesaw table.
+     */
+    fun getSelection(args: List<String>, table: Table): Selection
+}
+
+/**
+ * Implements the most basic filter.
+ * matches(columnName, regex)
+ */
+class Matches : JurisdictionalFilter {
+    override val name = "matches"
+
+    override fun getSelection(args: List<String>, table: Table): Selection {
+        if (args.size != 2) error("Expecting two args to filter $name:  (columnName, regex)")
+        val columnName = args[0]
+        val pattern = args[1]
+        return table.stringColumn(columnName).matchesRegex(pattern)
+    }
+}
+
+/**
+ * This may or may not be a unicorn.
+ */
+class FilterByCounty : JurisdictionalFilter {
+    override val name = "filterByCounty"
+
+    // @todo need tons of error checking.
+    override fun getSelection(args: List<String>, table: Table): Selection {
+        if (args.size != 2) error("Expecting two args to filter $name:  (TwoLetterState, County)")
+        val patientState = table.stringColumn("standard.patient_state")
+            ?: error("Unable to filterByCounty:  column standard.patient_state not found.")
+        val patientCounty = table.stringColumn("standard.patient_county")
+            ?: error("Unable to filterByCounty:  column standard.patient_county not found.")
+        val facilityState = table.stringColumn("standard.ordering_facility_state")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_state not found.")
+        val facilityCounty = table.stringColumn("standard.ordering_facility_county")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_county not found.")
+
+        // Try to be very loose on county matching.   Anything with the county name embedded is ok.
+        val countyRegex = "(?i).*${args[1]}.*"
```

Comment:
If we implement strong validation on county names, then we can make this a much tighter matching.